### DISCUSSION TITLE
ONVIF Conformance test failing due to responses

### DISCUSSION THREAD

**Discussion by sergio-mercado** [onvif/testspecs#83] [Q&A] [answered]
Hello everyone,

I'm looking for some help debugging an ONVIF profile S conformance test failing. I see most tests failing, and when I take a look at the pcap for the failed tests, it seems like we get the expected response code and tag. I have attached screenshots from the pcap and ONFIV test tool, as well as the pcap.

Feature list XML was added to all the cameras and and the three cameras were identified by the tool. The version of the tool which was used was:
24.06 rev. 84

For example, taking a look at the test 'CAPABILITIES-2 GET CAPABILITIES', S1 test passes and S2 and S3 fail. Taking a closer look this test has failed due to a missing **200 OK** response code and **<getCapabilitiesResponse** tag, as shown on the Diagnostics of the reporting tool, the frame analyzed for this test is frame 375. But that frame contains a **200 OK** code, and the GetCapabilitiesResponse tag as well.

Any help is appreciated.
[Onvif-Aug22-test3.pcapng.zip](https://github.com/user-attachments/files/16852174/Onvif-Aug22-test3.pcapng.zip)
![Screenshot 2024-08-23 at 10 52 40 AM](https://github.com/user-attachments/assets/d74006b4-9032-4212-b80d-ba8d0be4a357)
![Reporting](https://github.com/user-attachments/assets/70d4c16d-c2b8-4b91-93b4-efaec88ba427)
![Diagnostics](https://github.com/user-attachments/assets/87604bbd-3a74-4039-bfaa-d519b9b6f1c2)

**Comment by madhurao68** [comment]
Hello,
We are investigating and shall keep you posted. Thanks for your patience.

**Comment by madhurao68** [comment]
Hello,
Kindly let us know the application (with version) you used to collect the logs provided. Thanks for your patience.

**Reply by sergio-mercado** [reply]
Version used: 24.06 rev. 84

**Comment by madhurao68** [comment, marked as answer]
Hi,

Our test tool vendor investigated with the logs, and it seems that the cameras are connected with a virtual network.

![image](https://github.com/user-attachments/assets/c04f4cf1-a5b0-4ea6-9439-9b1b2f6c71c5)

We can provide a fix, but it will be available only in the next version of the tool v24.12. But as a workaround we suggest recording all tests without virtual networks. Thanks.

**Reply by sergio-mercado** [reply]
Thank you for investigating @madhurao68, I will attempt the test again without the VLAN and circle back with an update.

**Reply by sergio-mercado** [reply]
For anyone running into this issue, you can also use tcprewrite to delete vlan tags as a workaround, while the fix is being developed:
`tcprewrite --enet-vlan=del`

**Comment by madhurao68** [comment]
Hello,
The parsing issue caused by VLAN is fixed and is included in the latest intermediate release v24.09.  Appreciate if you could test and provide your feedback. Thanks.

Intermediate release link:  https://github.com/onvif/ctt/releases/tag/CTT24.09

**Reply by sergio-mercado** [reply]
Thank you. Looks like I don't have access, as I'm getting a 404 when trying to open the link.

**Comment by madhurao68** [comment]
Hi
Intermediate build link: https://drive.google.com/drive/folders/1z6cNpIdTW1tLZ_BwySMW3uVb0yVm_QMs?usp=sharing
